File: campeonato-backend/src/main/java/com/idat/campeonato_backend/entity/Partido.java
package com.idat.campeonato_backend.entity;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "partidos")
public class Partido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "equipo_local_id", nullable = false)
    @JsonIgnore
    private Equipo equipoLocal;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "equipo_visitante_id", nullable = false)
    @JsonIgnore
    private Equipo equipoVisitante;

    @Column(nullable = false)
    private Integer golesLocal;

    @Column(nullable = false)
    private Integer golesVisitante;

    @Column(nullable = false)
    private LocalDate fecha;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_campeonato", nullable = false)
    @JsonIgnore
    private Campeonato campeonato;
}

File: campeonato-frontend/src/app/models/partido.ts

export interface Partido {
    id?: number;
    equipoLocalId: number;
    equipoVisitanteId: number;
    golesLocal: number;
    golesVisitante: number;
    fecha: string;
    idCampeonato: number;
}

export interface PartidoResponse {
    id: number;
    equipoLocalNombre: string;
    equipoVisitanteNombre: string;
    golesLocal: number;
    golesVisitante: number;
    fecha: string;
    campeonatoNombre: string;
}

export interface PosicionTabla {
    idEquipo: number;
    nombreEquipo: string;
    pj: number;
    g: number;
    e: number;
    p: number;
    gf: number;
    gc: number;
    pts: number;
    get diferenciaGoles(): number; 
}

